src/python/deepgnn/graph_engine/data/reddit.py

```diff
@@ -0,0 +1,147 @@
+
+    Sampled 50 large communities and built a post-to-post graph, connecting posts if the same user comments
+    on both. For features, we use off-the-shelf 300-dimensional GloVe CommonCrawl word vectors.
+
+    References:
+        http://snap.stanford.edu/graphsage/#datasets
+        https://github.com/williamleif/GraphSAGE/blob/master/graphsage/utils.py
+
+    Graph Statistics:
+    - Nodes: 232,965
+    - Edges: 114,618,780 * edge_downsample_pct
+    - Split: (Train: 152410, Valid: 23699, Test: 55334)
+    - Node Label Dim: 50 (id:0)
+    - Node Feature Dim: 300 (id:1)
+    """
+
+    def __init__(
+        self,
+        output_dir: str = None,
+        edge_downsample_pct: float = 0.1,
+        num_partitions: int = 2,
+    ):
+        """
+        Initialize Reddit dataset.
+
+        Args:
+          output_dir (string): file directory for graph data.
+          edge_downsample_pct (float, default=.1): Percent of edges to use, default reddit graph has 100M edges.
+          num_partitions (int, default=2): Number of partitions
+        """
+        self._edge_downsample_pct = edge_downsample_pct
+        self._num_partitions = num_partitions
+        self.url = "https://snap.stanford.edu/graphsage/reddit.zip"
+        self.GRAPH_NAME = "reddit"
+        self.output_dir = output_dir
+        if self.output_dir is None:
+            self.output_dir = os.path.join("/tmp/", self.GRAPH_NAME)
+        load_graph_output = self._load_raw_graph(self.output_dir)
+        self._build_graph(self.output_dir, load_graph_output)
+        super(PPI, self).__init__(
+            path=self.output_dir, partitions=list(range(self._num_partitions))
+        )
+
+    def _load_raw_graph(self, output_dir: str):
+        raw_data_dir = os.path.join(output_dir, "raw")
+        data_dir = os.path.join(raw_data_dir, self.GRAPH_NAME)
+        fname = f"{data_dir}.zip"
+        download_file(self.url, raw_data_dir, f"{self.GRAPH_NAME}.zip")
+        with zipfile.ZipFile(fname) as z:
+            z.extractall(raw_data_dir)
+
+        id_map = json.load(open(os.path.join(data_dir, "reddit-id_map.json")))
+        id_map = {k: v for i, (k, v) in enumerate(id_map.items())}
+
+        g = json.load(open(os.path.join(data_dir, "reddit-G.json")))
+        nodes = []
+        nodes_type = []
+        train_nodes = []
+        NODE_TYPE_ID = {"train": 0, "val": 1, "test": 2}
+        for nid, node in enumerate(g["nodes"]):
+            nid = id_map[node["id"]]
+            if node["test"]:
+                ntype = NODE_TYPE_ID["test"]
+            elif node["val"]:
+                ntype = NODE_TYPE_ID["val"]
+            else:
+                ntype = NODE_TYPE_ID["train"]
+                train_nodes.append(nid)
+            nodes.append(nid)
+            nodes_type.append(ntype)
+
+        train_neighbors: Dict = {nid: [] for nid in id_map.values()}
+        other_neighbors: Dict = {nid: [] for nid in id_map.values()}
+
+        # edges
+        np.random.seed(0)
```

Comment:
nit: you probably want to put this seed to main function and let callers to control seeds.